### PR TITLE
Adjust ontology types GQL typedefs

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR changes the GQL types of the ontology types, such that they make use of the subgraph library's definition of the types. 
This by extension fixes the ability to create/update entity types and property types.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1203384018265719/f) _(internal)_

## 🚫 Blocked by

- Based on https://github.com/hashintel/hash/pull/1383

## 🔍 What does this change?

- See commits

## ⚠️ Known issues

The `*WithoutId` types are using the shared library, this could be in the subgraph package instead for consistency?

## 🐾 Next steps

- Continuing the parent task migrating knowledge.

## 🛡 What tests cover this?

- None yet.

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Change loggedInAndSignedUp to the following body:
```ts
  next(
    obj,
    args,
    {
      ...ctx,
      userModel: { entityId: "00000000-0000-0000-0000-000000000000" },
    } as any as LoggedInGraphQLContext,
    info,
  );
```
1.  Start external services + backend
1.  try to create/update either entity or property type.

## 📹 Demo

These screenshots shows creation/updating working

Creating entity type:
image

Updating a property type:
image

### CODE REVIEW

File: packages/hash/api/src/graphql/typeDefs/ontology/data-type.typedef.ts [hunk 1]
```diff
@@ -1,29 +1,9 @@
...
-    accountId: ID!
-      @deprecated(reason: "accountId is deprecated. Use ownedById instead.")
-    """
-    The data type
-    """
-    dataType: DataType!
-  }
+  # scalar DataType
+  # scalar DataTypeWithoutId
+  # scalar DataTypeWithMetadata
```
Comment (thehabbos007) [inline packages/hash/api/src/graphql/typeDefs/ontology/data-type.typedef.ts:6]: Datatype resolvers only return subgraphs currently

File: packages/hash/api/src/graphql/resolvers/middlewares/loggedInAndSignedUp.ts [hunk 2]
```diff
@@ -8,4 +8,13 @@ export const loggedInAndSignedUp: ResolverMiddleware<
...
+  // loggedIn(signedUp(next))(obj, args, ctx, info);
+  next(
+    obj,
+    args,
+    {
+      ...ctx,
+      userModel: { entityId: "00000000-0000-0000-0000-000000000000" },
+    } as any as LoggedInGraphQLContext,
+    info,
+  );
```
Comment (benwerner01) [inline packages/hash/api/src/graphql/resolvers/middlewares/loggedInAndSignedUp.ts:20]: Did this accidentally get committed?

**Review by benwerner01 — CHANGES_REQUESTED** [review]
Thanks Ahmad, changes look good to me and are working locally but I think I've spotted two blocking things worth addressing in this PR.

File: packages/hash/api/src/graphql/resolvers/middlewares/loggedInAndSignedUp.ts (hunk 2 above)
Comment (thehabbos007) [reply packages/hash/api/src/graphql/resolvers/middlewares/loggedInAndSignedUp.ts:20]: Yes, my bad, must have slipped in as I rebased

Comment (thehabbos007) [reply packages/hash/api/src/graphql/resolvers/middlewares/loggedInAndSignedUp.ts]: Undone in a94a5695fbd4fa5b77731d5c57609e717876770c

File: packages/hash/integration/codegen.yml [hunk 3]
```diff
@@ -15,13 +15,14 @@ generates:
...
         DataTypeWithoutId: "@hashintel/hash-shared/graphql/types#DataTypeWithoutId"
-        PropertyTypeWithoutId: "@hashintel/hash-shared/graphql/types#PropertyTypeWithoutId"
+        DataTypeWithMetadata: "@hashintel/hash-subgraph#DataTypeWithMetadata"
+        EntityType: "@blockprotocol/type-system-web#EntityType"
         EntityTypeWithoutId: "@hashintel/hash-shared/graphql/types#EntityTypeWithoutId"
-        LinkTypeWithoutId: "@hashintel/hash-shared/graphql/types#LinkTypeWithoutId"
+        EntityTypeWithMetadata: "@hashintel/hash-subgraph#EntityTypeWithMetadata"
+        PropertyType: "@blockprotocol/type-system-web#PropertyType"
+        PropertyTypeWithoutId: "@hashintel/hash-shared/graphql/types#PropertyTypeWithoutId"
+        PropertyTypeWithMetadata: "@hashintel/hash-subgraph#PropertyTypeWithMetadata"
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/integration/codegen.yml]: I think that would be great. I can add a todo, but will just check if there is an out-of-the box solution I can link to for htis

[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/integration/codegen.yml]: Added todo in 13b1ee7f50eba9c9d8d74270b410fd0ee19377ae
I found https://the-guild.dev/graphql/codegen/docs/config-reference/codegen-config which may be useful!

[the remark this answers is not in the document]
Comment (benwerner01) [reply packages/hash/integration/codegen.yml]: Thanks :)

File: packages/hash/shared/src/graphql/types.ts [hunk 4]
```diff
@@ -5,10 +5,9 @@ import {
   EntityType,
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/shared/src/graphql/types.ts:5]: Done in 13b1ee7f50eba9c9d8d74270b410fd0ee19377ae

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**